.gitignore

```diff
@@ -18,6 +18,9 @@
 
 # Ignore application configuration
 /config/application.yml
+/public
+Rakefile
+Gemfile.lock
```

Comment:
Yes I agree with @dblock, what I meant earlier is that your changes to these files should not be in the commits.

To get ride of the already committed `/public/*` files, try deleting them, committing those deletes, and then [squashing the commits](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git).